doc/holochain_101/src/zome/read_and_write.md

```diff
@@ -1 +1,154 @@
+3. The only time this does not apply is for the very first header, for which there is no previous header hash, and a value of `0` is used
+
+Other data contained in the headers includes (but isn't limited to):
+- the public key/address, and the cryptographically signed signature of the author
+- a timestamp conveying the moment of creation of the header
+- the hash of the "entry" being created
+- the "entry type" of the "entry" being created
+
+The meaning and role of entries is covered next.
+
+### Entries
+
+Entries are elements of data which get persisted to the same storage as Headers, yet have a different nature. They represent the substance of the data that is of interest to the end users of Holochain, storing records such as transactions, social media posts, or whatever else app developers come up with. The hash of each entry is stored and referenced in its corresponding header. Every local source chain should have the exact same number of Entries and Headers.
+
+All Entries have a type. Entries can be "system" Entries, or "app" Entries. System Entries are Entries integral to the proper functioning of Holochain. Importantly, there are system Entries written to each local source chain, at the time of its initialization, or "genesis".
+
+#### System Entries
+These include firstly, the DNA itself. The DNA is written as the first entry into each local source chain. Importantly, this is what gives everyone totaly confidence in everyone elses chains: the fact that you all know you started out by running the same code (otherwise, the hash of the initial DNA entry would be different). There will be more discussion on this later.
+
+The second Entry to be written to the chain is information about the agent using this source chain. That includes their "public key" which is also their address on the network.
+
+There are also entries created relating to the "Capabilities" features of Holochain, but those will be covered elsewhere.
+
+#### App Entries
+Next, there are the "App" Entries. The definitions for these entry types, and how to validate them, are in the hands of developers. When end users are attempting actions which should author data, the data will first be checked against the validation rules supplied by the DNA developer. The possibilities for App Entry type definitions are endless. Defining App Entry types is how you setup the schemas, data structures, and data integrity of your DNA.
+
+#### Entry Sharing
+Not every Entry must be shared with other people running the same DNA. There are two (and may be more later) levels of sharing for App Entries defined: `public` and `private`.
+
+With `public`, all Entries of that type will get propogated to the other peers running the same DNA, so that they will hold a copy as well, and make your data more resilient, and so that it is available even when you're offline.
```

Comment:
```suggestion
With `public`, all Entries of that type will get propagated to the other peers running the same DNA, so that they will hold a copy as well, and make your data more resilient, and so that it is available even when you're offline.
```